p2p/src/swarm.rs

```diff
@@ -0,0 +1,645 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+#![allow(unused, dead_code)]
+#![cfg(not(loom))]
+
+use crate::{
+    error::{self, P2pError},
+    event,
+    net::{self, ConnectivityService, NetworkService},
+    peer::{Peer, PeerRole},
+};
+use common::chain::ChainConfig;
+use futures::FutureExt;
+use logging::log;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+use tokio::sync::mpsc;
+
+const MAX_ACTIVE_CONNECTIONS: usize = 32;
+
+#[allow(unused)]
+#[derive(Debug, PartialEq, Eq)]
+enum PeerState {
+    /// Peer is handshaking
+    Handshaking,
+
+    /// Peer is ready for flooding/syncing
+    Active,
+}
+
+#[allow(unused)]
+#[derive(Debug)]
+struct PeerContext<T>
+where
+    T: NetworkService,
+{
+    /// Unique peer ID
+    id: T::PeerId,
+
+    /// Peer sta
+    state: PeerState,
+
+    /// Channel for communication with the peer
+    tx: mpsc::Sender<event::Event>,
+}
+
+#[allow(unused)]
+enum PeerAddrInfo<T>
+where
+    T: NetworkService,
+{
+    Raw {
+        /// Hashset of IPv4 addresses
+        ip4: HashSet<Arc<T::Address>>,
+
+        /// Hashset of IPv6 addresses
+        ip6: HashSet<Arc<T::Address>>,
+    },
+}
+
+pub struct SwarmManager<T>
```

Comment:
Isn't the concept of "swarm" libp2p specific, which means the backend object that controls the connections? Are we deciding here to call any backend object a swarm? The name swarm to me is very confusing... not sure this is the best label. While this isn't a big deal, maybe in the worst case, we should write a little comment explaining what a swarm is, since it's part of our interface.

Reply:
I think the word swarm itself is a very good description for a collection of peers. It's not used only in libp2p but also, e.g., in BitTorrent and it basically just describes a set of peer we know about/are connected to. The problem I have with libp2p's use of the word swarm is that the object they call `Swarm` is not a swarm, but rather a libp2p top-level object that handles everything from swarm management to mDNS peer discovery and floodsub pub/sub functionality. So the word swarm is not descriptive of what it actually does but here our use of the word is IMO very descriptive because this is a manager object that handles all swarm-related functionality and that's all. I'll add a comment about swarm though.